jfix-stdlib-concurrency/src/test/kotlin/ru/fix/stdlib/concurrency/events/EventReducerTest.kt

```diff
@@ -0,0 +1,154 @@
+package ru.fix.stdlib.concurrency.events
+
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import ru.fix.aggregating.profiler.NoopProfiler
+import java.util.concurrent.ArrayBlockingQueue
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.ReentrantLock
+
+private const val INVOCATION_TIMEOUT_MS = 500L
+
+internal class EventReducerTest {
+
+    @Test
+    fun `WHEN handle single time THEN handler invoked one time`() {
+        val invokes = ArrayBlockingQueue<Any>(1)
+        EventReducer.lastEventWinReducer<Any>(profiler = NoopProfiler(), handler = {
+            invokes.put(it!!)
+        }).use {
+            it.start()
+            it.handleEvent(Any())
+            assertNotNull(invokes.poll(INVOCATION_TIMEOUT_MS, TimeUnit.MILLISECONDS))
+            assertNull(invokes.poll(INVOCATION_TIMEOUT_MS, TimeUnit.MILLISECONDS))
+        }
+    }
+
+    @Test
+    fun `WHEN reducer triggered several times at once AND handler is slow THEN handler invoked one or two times`() {
+        val eventsQuantity = 10
+        val invokes = ArrayBlockingQueue<Any>(2)
+        val awaitingEventsFromAllThreadsLatch = CountDownLatch(eventsQuantity)
+        EventReducer.lastEventWinReducer<Any>(profiler = NoopProfiler(), handler = {
+            invokes.put(it!!)
+            awaitingEventsFromAllThreadsLatch.await() //simulate slow handler
+        }).use { reducer ->
+            reducer.start()
+
+            val executor = Executors.newFixedThreadPool(eventsQuantity)
+            repeat(eventsQuantity) {
+                executor.execute {
+                    reducer.handleEvent(Any())
+                    awaitingEventsFromAllThreadsLatch.countDown()
+                }
+            }
+
+            assertNotNull(invokes.poll(INVOCATION_TIMEOUT_MS, TimeUnit.MILLISECONDS))
+            invokes.poll(INVOCATION_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+            assertNull(invokes.poll(INVOCATION_TIMEOUT_MS, TimeUnit.MILLISECONDS))
+
+            executor.shutdown()
+        }
+    }
+
+    @Test
+    fun `WHEN handle event after invoking handler THEN handler will be invoked after completing`() {
```

Comment:
there is no shutdown here, why `completing` word in the test; 
Maybe name test like that: 
`publish event when handler is still working, handler will be invoked again and event will be handled`

Reply:
I removed this test due to removing handler from constructor parameters